### PR TITLE
PWGHF: add THn for Lc and Sc.

### PR DESCRIPTION
**Pull request by mfaggin**

### CODE REVIEW

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 1]
```diff
@@ -542,17 +611,23 @@ struct HfTaskSigmac {
...
 
           /// Fill the histograms for reconstructed Σc0,++ signal
           registry.fill(HIST("MC/reconstructed/hPtSc0PlusPlusSig"), ptSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPtGenSc0PlusPlusSig"), ptGenSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hEtaSc0PlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiSc0PlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSig"), deltaMass, ptSc);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigNonPrompt"), deltaMass, ptSc); // Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:626]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 2]
```diff
@@ -562,8 +637,16 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0PlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0PlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0,++ signal
+          if (origin == 1) {
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:642]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 3]
```diff
@@ -562,8 +637,16 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0PlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0PlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:640]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 4]
```diff
@@ -578,8 +661,11 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaSc0Sig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiSc0Sig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassSc0Sig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0SigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0SigNonPrompt"), deltaMass, ptSc, channel); // Σc0 signal
+          if (origin == 1) {
+            registry.fill(HIST("MC/reconstructed/hDeltaMassSc0SigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:666]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 5]
```diff
@@ -589,17 +675,23 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0Sig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0Sig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0Sig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0SigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0SigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0 signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:678]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 6]
```diff
@@ -589,17 +675,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0SigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:680]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 7]
```diff
@@ -589,17 +675,23 @@ struct HfTaskSigmac {
...
 
           /// Fill the histograms for reconstructed Σc0,++ signal
           registry.fill(HIST("MC/reconstructed/hPtSc0PlusPlusSig"), ptSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPtGenSc0PlusPlusSig"), ptGenSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hEtaSc0PlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiSc0PlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigNonPrompt"), deltaMass, ptSc, channel); // Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:690]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 8]
```diff
@@ -589,17 +675,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:692]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 9]
```diff
@@ -609,8 +701,16 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0PlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0PlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:704]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 10]
```diff
@@ -609,8 +701,16 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:706]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 11]
```diff
@@ -650,8 +752,11 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaScPlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiScPlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigNonPrompt"), deltaMass, ptSc, channel); // Σc0 signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:755]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 12]
```diff
@@ -650,8 +752,11 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:757]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 13]
```diff
@@ -661,17 +766,23 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromScPlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromScPlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0 signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:769]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 14]
```diff
@@ -661,17 +766,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:771]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 15]
```diff
@@ -661,17 +766,23 @@ struct HfTaskSigmac {
...
 
           /// Fill the histograms for reconstructed Σc0,++ signal
           registry.fill(HIST("MC/reconstructed/hPtSc0PlusPlusSig"), ptSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPtGenSc0PlusPlusSig"), ptGenSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hEtaSc0PlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiSc0PlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigNonPrompt"), deltaMass, ptSc, channel); // Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:781]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 16]
```diff
@@ -661,17 +766,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:783]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 17]
```diff
@@ -681,8 +792,16 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0PlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0PlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:795]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 18]
```diff
@@ -681,8 +792,16 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:797]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 19]
```diff
@@ -697,8 +816,11 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaScPlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiScPlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigNonPrompt"), deltaMass, ptSc, channel); // Σc0 signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:819]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 20]
```diff
@@ -697,8 +816,11 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassScPlusPlusSigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:821]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 21]
```diff
@@ -708,17 +830,23 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromScPlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromScPlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0 signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:833]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 22]
```diff
@@ -708,17 +830,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromScPlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:835]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 23]
```diff
@@ -708,17 +830,23 @@ struct HfTaskSigmac {
...
 
           /// Fill the histograms for reconstructed Σc0,++ signal
           registry.fill(HIST("MC/reconstructed/hPtSc0PlusPlusSig"), ptSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPtGenSc0PlusPlusSig"), ptGenSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hEtaSc0PlusPlusSig"), etaSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiSc0PlusPlusSig"), phiSc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSig"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigNonPrompt"), deltaMass, ptSc, channel); // Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:845]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 24]
```diff
@@ -708,17 +830,23 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassSc0PlusPlusSigPrompt"), deltaMass, ptSc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:847]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 25]
```diff
@@ -728,14 +856,44 @@ struct HfTaskSigmac {
           registry.fill(HIST("MC/reconstructed/hEtaLcFromSc0PlusPlusSig"), etaLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hPhiLcFromSc0PlusPlusSig"), phiLc, origin, channel);
           registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSig"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
-          registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigNonPrompt"), deltaMass, ptLc, channel); // Λc+ ← Σc0,++ signal
+          if (origin == 1) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:859]:
```suggestion
          if (origin == RecoDecay::OriginType::Prompt) {
```

File: PWGHF/D2H/Tasks/taskSigmac.cxx [hunk 26]
```diff
@@ -728,14 +856,44 @@ struct HfTaskSigmac {
...
+            registry.fill(HIST("MC/reconstructed/hDeltaMassLcFromSc0PlusPlusSigPrompt"), deltaMass, ptLc, channel);
+          } else if (origin == 2) {
```
Comment (fgrosa) [inline PWGHF/D2H/Tasks/taskSigmac.cxx:861]:
```suggestion
          } else if (origin == RecoDecay::OriginType::NonPrompt) {
```

**Comment by mfaggin** [comment]
> Hi @mfaggin I just have a minor comment (repeated many times). As a general comment maybe it could be better to group all the histograms for prompt/nonprompt production not to repeat many times the same if.

Thanks @fgrosa , I've followed your suggestions in the latest commit

**Review by fgrosa — APPROVED** [review]

**Pull request merged**